src/storage-operators/src/s3_oneshot_sink.rs

```diff
@@ -229,6 +231,25 @@ where
                     }
                 };
 
+                // Confirm we have DeleteObject permissions before proceeding by trying to
+                // delete a known non-existent file and checking if an error is returned
```

Comment:
Worth a comment saying that s3 does in fact give a permission error even when trying to delete an object that doesn't exist (and that in fact, it does not error)